### PR TITLE
chore(broker): add more specific logs for taking snapshot

### LINKED ISSUE #4553: Improve log message when failing to obtain a pending snapshot
**Description**

We currently log "Failed to obtain pending snapshot for position {}" when the async snapshot director cannot obtain a pending snapshot ref from the snapshot store. This can happen for normal reasons, such as us attempting to take a snapshot for an index we already have, for example. This log can be confusing as it sounds like an error - which it can be, but not always. We should improve this and provide the reason, as well as pick the correct log level based on the reason (e.g. if expect log it as debug or info, if unexpected log it as error).

@Zelldon maybe this is something you can do as part of refactoring the snapshot stuff

### LINKED ISSUE #5142: Failure message why snapshot was not take is to generic
**Description**

We currently have multiple failure cases in the code why we can't take a snapshot but often we only see
`Failed to obtain a pending snapshot directory for position 1063013`, which is way to generic. It is totally unclear for the reader what excatly has failed, or what was expected etc. We should improve this error message, this would help investigating a lot.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Add more specific log messages for taking a snapshot. Makes more clear when something went wrong with help of more context.

## Related issues

closes https://github.com/zeebe-io/zeebe/issues/5142
closes https://github.com/zeebe-io/zeebe/issues/4553

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to backport the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the release announcement

### CODE REVIEW

**Comment by Zelldon** [comment]
I know this fix's not really a bug, but maybe still worth to back port it @npepinpe , since I it makes debugging easier. wdyt?

File: broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java [hunk 1]
```diff
@@ -159,26 +166,6 @@ boolean isDbOpened() {
     return db != null;
   }
 
-  private void createSnapshot(final TransientSnapshot snapshot) {
```
Comment (deepthidevaki) [inline broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java:162]: Why is this removed?

File: broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java [hunk 2]
```diff
@@ -71,29 +71,36 @@ public StateControllerImpl(
...
-        optionalIndexed
-            .filter(indexed -> indexed.index() != previousSnapshotIndex)
-            .map(
-                indexed ->
-                    store.newTransientSnapshot(
-                        indexed.index(),
-                        indexed.entry().term(),
-                        WallClockTimestamp.from(System.currentTimeMillis())));
-
-    optTransientSnapshot.ifPresent(this::createSnapshot);
```
Comment (deepthidevaki) [inline broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java:95]: I think this was removed by mistake :slightly_smiling_face: . This is probably why tests are failing.

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
Thanks :+1: I have some minor suggestions on the log messages. You can apply them if it make sense. 
The tests are failing. See my comment below.

**Comment by npepinpe** [comment]
These are small changes which don't change the behaviour of the system and help us better diagnose issues, so I think it's a good candidate to backport (as long as it's easy to backport) :+1:

File: broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java [hunk 3]
```diff
@@ -71,29 +71,36 @@ public StateControllerImpl(
   @Override
   public Optional<TransientSnapshot> takeTransientSnapshot(final long lowerBoundSnapshotPosition) {
     if (!isDbOpened()) {
+      LOG.warn(
+          "Expected to take snapshot for lower bound position {}, but database was closed.",
```
[the remark this answers is not in the document]
Comment (Zelldon) [reply broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java:75]: Actually I wrote not last processed since it is not really true. Do you think it is confusing to write lower bound?

File: broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java (hunk 2 above)
Comment (Zelldon) [reply broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java:95]: Ups :D yeah thanks will revert that :see_no_evil:

**Comment by Zelldon** [comment]
Hey @deepthidevaki thanks for the fast review. I applied your hints, please have another look.

**Review by deepthidevaki — APPROVED** [review]
:+1: Thanks
